expr/expr.go

```diff
@@ -3341,10 +3342,12 @@ func EvalExpr(e *expr, from, until int32, values map[MetricRequest][]*MetricData
 				current = fmt.Sprintf("%s %s", current, unit)
 			}
 
-			a.Name = proto.String(fmt.Sprintf("%s Current: %s Max: %s Min: %s", *a.Name, current, max, min))
+			r := *a
+			r.Name =proto.String(fmt.Sprintf("%s Current: %s Max: %s Min: %s", *a.Name, current, max, min))
+			metrics = append(metrics, &r)
```

Comment:
It's more than just the tests.  We only fetch a single metric once, even if it's used in more than one expression.  If we modify a metric value here, then the other expressions will evaluate incorrectly.